frontend/src/components/pages/RegistrantExperience/PersonalInfo/ContactCard.tsx

```diff
@@ -0,0 +1,225 @@
+import React, { useEffect, useState } from "react";
+import {
+  Box,
+  Divider,
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+  Heading,
+  Input,
+  Spacer,
+  Text,
+  Textarea,
+  Wrap,
+  WrapItem,
+} from "@chakra-ui/react";
+import {
+  PersonalInfoActions,
+  PersonalInfoReducerDispatch,
+} from "../../../../types/PersonalInfoTypes";
+import { EmergencyContact } from "../../../../types/CamperTypes";
+import RequiredAsterisk from "../../../common/RequiredAsterisk";
+import {
+  checkEmail,
+  checkFirstName,
+  checkLastName,
+  checkPhoneNumber,
+  checkRelationToCamper,
+} from "./personalInfoReducer";
+
+type ContactCardProps = {
+  nextBtnRef: React.RefObject<HTMLButtonElement>;
+  contact: EmergencyContact;
+  contactId: number;
+  setPersonalInfo: (action: PersonalInfoReducerDispatch) => void;
+};
+
+const ContactCard = ({
+  nextBtnRef,
+  contact,
+  contactId,
+  setPersonalInfo,
+}: ContactCardProps): React.ReactElement => {
+  const [isFirstNameInvalid, setIsFirstNameInvalid] = useState<boolean>(false);
+  const [isLastNameInvalid, setIsLastNameInvalid] = useState<boolean>(false);
+  const [isEmailInvalid, setIsEmailInvalid] = useState<boolean>(false);
+  const [isPhoneNumberInvalid, setIsPhoneNumberInvalid] = useState<boolean>(
+    false,
+  );
+  const [isRelationInvalid, setIsRelationInvalid] = useState<boolean>(false);
+
+  useEffect(() => {
+    const updateFormErrorMsgs = () => {
+      console.log(
+        "contact",
+        checkFirstName(contact.firstName),
+        checkLastName(contact.lastName),
+        checkEmail(contact.email),
+      );
+      if (!checkFirstName(contact.firstName)) setIsFirstNameInvalid(true);
+      if (!checkLastName(contact.lastName)) setIsLastNameInvalid(true);
+      if (!checkEmail(contact.email)) setIsEmailInvalid(true);
+      if (!checkPhoneNumber(contact.phoneNumber)) setIsPhoneNumberInvalid(true);
+      if (!checkRelationToCamper(contact.relationshipToCamper))
+        setIsRelationInvalid(true);
+    };
+
+    if (nextBtnRef && nextBtnRef.current) {
+      // Passing the same reference
+      nextBtnRef.current.addEventListener("click", updateFormErrorMsgs);
+    }
+
+    return () => {
+      // Passing the same reference
+      if (nextBtnRef && nextBtnRef.current) {
+        nextBtnRef.current.removeEventListener("click", updateFormErrorMsgs);
+      }
+    };
+  }, [contact]);
+
+  return (
+    <Box boxShadow="lg" rounded="xl" borderWidth={1} width="100%">
+      <Box backgroundColor="#FFFFFF" rounded="xl">
+        <Heading textStyle="displayLarge">
+          <Text
+            py={6}
+            px={{ sm: "5", lg: "20" }}
+            textStyle={{ sm: "xSmallBold", lg: "displayLarge" }}
+          >
+            {contactId === 0 ? "Primary Contact" : "Secondary Contact"}{" "}
```

Comment:
can we also rename this to `contactIndex` ?

Reply:
Yep!